extensions/wikia/LinkSuggest/LinkSuggestLoader.class.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * @class LinkSuggestLoader
+ * Proper loader for LinkSuggest
+ * @author TK-999
+ */
+class LinkSuggestLoader {
+	/**
+	 * @var array $selectors : Array of jQuery-style selectors to apply LinkSuggest to
+	 */
+	private $selectors = [];
+
+	/**
+	 * @var bool $userWantsLinkSuggest : Whether the current user has enabled linkSuggest
+	 */
+	private $userWantsLinkSuggest = false;
+
+	private static $instance = null;
+
+	private function __construct() {
+		$app = F::app();
+		$this->userWantsLinkSuggest = !( $app->wg->User->getGlobalPreference( 'disablelinksuggest' ) );
+
+		// only register hook if the user has enabled LinkSuggest
+		if ( $this->userWantsLinkSuggest ) {
+			$app->registerHook( 'BeforePageDisplay', get_class( $this ), 'onBeforePageDisplay', [], false, $this );
```

Comment:
I worry that this can cause some unexpected behaviour that will be harder to debug,
I would register it always and in hook check if you need to perform any action
this way when something happens you know when the hook run and if it brakes something